fenics_adjoint/types/constant.py

```diff
@@ -16,9 +16,21 @@
 @register_overloaded_type
 class Constant(OverloadedType, backend.Constant):
     def __init__(self, *args, **kwargs):
+        annotate = annotate_tape(kwargs)
         super(Constant, self).__init__(*args, **kwargs)
         backend.Constant.__init__(self, *args, **kwargs)
 
+        if annotate:
+            if len(args) > 0 and isinstance(args[0], OverloadedType):
+                other = args[0]
+                if not isinstance(other, OverloadedType):
+                    other = create_overloaded_object(other)
```

Comment:
This seems to contradict the condition above.
Any idea about how to handle vector Constants?

Reply:
Ahh, yes. That is what I get for copy-pasting boilerplate code.

Reply:
I've added support for list and tuple inputs.